Doc/whatsnew/3.12.rst

```diff
@@ -993,6 +1004,10 @@ Deprecated
   (Contributed in :gh:`47146` by Petr Viktorin, based on
   earlier work by Alexander Belopolsky and Matthias Braun.)
 
+* :c:func:`PyErr_Fetch` and :c:func:`PyErr_Restore` are deprecated.
+  Use :c:func:`PyErr_GetRaisedException` and
+  :c:func:`PyErr_SetRaisedException` instead.
```

Comment:
(Contributed by Mark Shannon in https://github.com/python/cpython/issues/101578)